internal/controller/controllers/agent_controller.go

```diff
@@ -1022,6 +1027,35 @@ func (r *AgentReconciler) updateInventoryLabels(log logrus.FieldLogger, ctx cont
 	return nil
 }
 
+func (r *AgentReconciler) updateClusterDeploymentRefLabels(log logrus.FieldLogger, ctx context.Context, agent *aiv1beta1.Agent) error {
+	name := ""
+	namespace := ""
+
+	if agent.Spec.ClusterDeploymentName != nil {
+		name = agent.Spec.ClusterDeploymentName.Name
+		namespace = agent.Spec.ClusterDeploymentName.Namespace
+	}
+
+	changed := false
+	changed = setAgentLabel(log, agent, "agent-install.openshift.io/clusterdeployment-name", name) || changed
```

Comment:
Just realized that the maximum length for a label value is 63 characters, but a resource name can be up to 253.  Namespace can be up to 63 so I suggest we add a label only for the namespace - at least it will help us filter the clusterdeployments to down a namespace.  Unless you have other ideas?

Reply:
Good catch, would have caused us some trouble sooner or later... Let's just label by namespace, that should be good enough for start.